### PR TITLE
Added Title Tab to Window and Implemented Code Changes

### PR DESCRIPTION
**Pull request by nishant2253**
This pull request addresses the issue of missing title tabs in the window for each page in the DSOMM (DevSecOps Maturity Model) application. 

   **App.component.ts**:
        Added functionality to dynamically update the document title based on the current route.
        Utilized the Title service from @angular/platform-browser to set the document title.
        Implemented a method updateTitle() to update the document title with the appropriate page title.
        Subscribed to router events to listen for navigation end events and update the title accordingly.

   **App.component.spec.ts**:
        Updated tests to accommodate the changes made in the app.component.ts file.
        Added tests to ensure that the document title is updated correctly when navigating to different routes.
    **App-routing.module.ts**:
        Modified route configurations to include a data property for each route specifying the title of the respective page.

### CODE REVIEW

**Comment by 0x41head** [comment]
LGTM!  @nishant2253 can you link the issue to this PR ?

**Review by wurstbrot — APPROVED** [review]

**Comment by wurstbrot** [comment]
please fix linting

**Comment by wurstbrot** [comment]
@nishant2253 Please fix linting error. Please run the linting command `node_modules/.bin/eslint . --ext .js,.jsx,.ts,.tsx` locally

@0x41head Do you know if the the unit tests are generally broken?

**Comment by nishant2253** [comment]
> LGTM! @nishant2253 can you link the issue to this PR ?

@0x41head My apologies for the delay. You can find the corresponding issue linked to this pull request under issue #268.

Additionally, I've created a new pull request for your review, labeled as #285. Could you please review it

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css'],
 })
-export class AppComponent {
-  title = 'DSOMM';
+export class AppComponent implements OnInit {
+  title: string = '';
```
[the remark this answers is not in the document]
Comment (nishant2253) [reply src/app/app.component.ts:14]: @wurstbrot Apologies for late reply, I've fixed the previous ESLint checks and established a constant, DEFAULT_TITLE, to define the default title. Could you please review my new pull request #285

**Comment by 0x41head** [comment]
> @0x41head Do you know if the the unit tests are generally broken?

First time looking at the unit tests after a long time, need to investigate why it's showing so many errors.
Though the test cases [usually](https://github.com/devsecopsmaturitymodel/DevSecOps-MaturityModel/pull/284/checks) pass, so it seems the PR breaks something as well.

**Pull request closed without merging**